File: 0648-replace-words/0648-replace-words.cpp
class Solution {
public:
    string replaceWords(vector<string>& dictionary, string sentence) {
        unordered_set<string>mp;
        for(string s:dictionary) {
            mp.insert(s);
        }
        sentence += " ";
        string ans = "";
        for(int i=0; i<sentence.size(); i++) {
            string word = "";
            while(sentence[i] != ' ') {
                word += sentence[i];
                i++;
                if(mp.count(word)) {
                    while(sentence[i] != ' ') i++;
                }
            }
            ans += word;
            ans += " ";
        }
        ans.pop_back();
        return ans;
    }
};

File: 0648-replace-words/0648-replace-words.java
class Solution {
    public String replaceWords(List<String> dictionary, String sentence) {
        Set<String>st = new HashSet<>();
        for(String s:dictionary) {
            st.add(s);
        }
        StringBuilder sb = new StringBuilder();
        for(int i=0; i<sentence.length(); i++) {
            StringBuilder str = new StringBuilder();
            while(i < sentence.length() && sentence.charAt(i) != ' ') {
                str.append(sentence.charAt(i));
                i++;
                if(st.contains(str.toString())) {
                    while(i < sentence.length() && sentence.charAt(i) != ' ') i++;
                }
            }
            sb.append(str);
            sb.append(" ");
        }
        sb.deleteCharAt(sb.length() - 1);
        String ans = sb.toString();
        return ans;
    }
}